### PR TITLE
Inline spartan-secq dependency

### PR DESCRIPTION
**Pull request by lsankar4033**
addresses https://github.com/personaelabs/spartan-ecdsa/issues/6

Inlines @DanTehrani 's fork of spartan made to work on secq to this repo s.t. the circular dependency exists entirely within this repo.

`cargo build` + `cargo test` pass in both relevant packages

### CODE REVIEW

File: packages/Spartan-secq/.github/workflows/rust.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: Build and Test Spartan
```
Comment (DanTehrani) [inline packages/Spartan-secq/.github/workflows/rust.yml:1]: I agree that we should set up CI but if we are going to add a GitHub action can we add it in the root dir?

Comment (DanTehrani) [reply packages/Spartan-secq/.github/workflows/rust.yml:1]: Or we can do that later by removing this rust.yml and merging this PR first!

**Review by DanTehrani — CHANGES_REQUESTED** [review]
@lsankar4033  Commented on the CI setup that other parts look good! Also confirmed that all tests pass!

File: packages/Spartan-secq/.github/workflows/rust.yml (hunk 1 above)
Comment (lsankar4033) [reply packages/Spartan-secq/.github/workflows/rust.yml:1]: ya let's remove this `rust.yml` for now and add CI at the root dir later

**Review by DanTehrani — APPROVED** [review]

**Pull request merged**